src/common/NetworkResult.hpp

```diff
@@ -20,13 +25,30 @@ class NetworkResult
     /// Parses the result as json and returns the document.
     rapidjson::Document parseRapidJson() const;
     const QByteArray &getData() const;
-    int status() const;
 
-    static constexpr int timedoutStatus = -2;
+    /// The error code of the reply.
+    /// In case of a successful reply, this will be NoError (0)
+    Error error() const
+    {
+        return this->error_;
+    }
+
+    /// The HTTP status code if a request was made.
+    std::optional<uint16_t> status() const
+    {
+        return this->status_;
+    }
+
+    /// Formats the error:
+    /// without HTTP status: [Name]
+    /// with HTTP status: [HTTP status]
```

Comment:
...if a successful reply was received, or something, returns the http status
otherwise, returns the network error